src/pexshell.rs

```diff
@@ -87,7 +87,7 @@ impl<'a> PexShell<'a> {
     async fn api_request(
         &mut self,
         client: reqwest::Client,
-        config: &impl config::Provider,
+        config: &mut impl Provider,
```

Comment:
```suggestion
        config: &mut impl ConfigProvider,
```

There might be a few more of these